crates/matrix-sdk-indexeddb/src/state_store/mod.rs

```diff
@@ -432,7 +433,12 @@ struct PersistedQueuedEvent {
     // All these fields are the same as in [`QueuedEvent`].
     event: SerializableEventContent,
     transaction_id: OwnedTransactionId,
-    is_wedged: bool,
+
+    // Deprecated (from old format), now replaced with error field.
+    // Kept here for migration
+    is_wedged: Option<bool>,
```

Comment:
Does the conversion from bool to `Option<bool>` work? (i.e. is there a test for that?)

I think it may work because we're using JSON as the serialization format, but it's not guaranteed it would work in general with any serialization format, so I'd rather have a test for this please.

Reply:
added here https://github.com/matrix-org/matrix-rust-sdk/pull/4137/commits/93c452872cae5915fa97b7f8eed9c55150d7adcf